subsys/llext/llext_link.c

```diff
@@ -52,6 +52,73 @@ static size_t llext_file_offset(struct llext_loader *ldr, size_t offset)
 	return offset;
 }
 
+/*
+ * We increment use-cound every time a new dependent is added, and have to
+ * decrement it again, when one is removed. Ideally we should be able to add
+ * arbitrary numbers of dependencies, but using lists for this doesn't work,
+ * because multiple extensions can have common dependencies. Dynamically
+ * allocating dependency entries would be too wasteful. In this initial
+ * implementation we use an array of dependencies, if at some point we run out
+ * of array entries, we'll implement re-allocation.
+ * We add dependencies incrementally as we discover them, but we only ever
+ * expect them to be removed all at once, when their user is removed. So the
+ * dependency array is always "dense" - it cannot have NULL entries between
+ * valid ones.
+ */
+static int llext_dependency_add(struct llext *ext, struct llext *dependency)
+{
+	unsigned int i;
+
+	for (i = 0; i < ARRAY_SIZE(ext->dependency); i++) {
+		if (ext->dependency[i] == dependency)
+			return 0;
+
+		if (!ext->dependency[i]) {
+			ext->dependency[i] = dependency;
+			dependency->use_count++;
+
+			return 0;
+		}
+	}
+
+	return -ENOENT;
+}
+
+void llext_dependency_remove(struct llext *ext)
```

Comment:
Nit: This name sounds like a poor choice to me; something like `llext_release_dependencies`/`llext_detach_from_dependencies`/... (emphasis on **cies**, i.e., make it clear this drops refcount of *all extensions we depend on*) would be more appropriate IMO.

Reply:
@mathieuchopstm I prefer naming from objects to verbs, i.e. `irq_disable()` not `disable_irq()`. And singular over plural in most cases. So, the choice was deliberate, but if we want to emphasise, that it's removing all dependencies, we could make it `llext_dependency_remove_all()` or similar. And "remove" is chosen to counteract `llext_dependency_add()`. But we could use "release" too, I guess

Reply:
I see your point. +1 on `llext_dependency_remove_all()`.